tsdb/engine/tsm1/engine_test.go

```diff
@@ -1819,28 +1750,168 @@ func BenchmarkEngine_WritePoints_Parallel(b *testing.B) {
+			Ascending: true,
+			StartTime: influxql.MinTime,
+			EndTime:   influxql.MaxTime,
+		},
+	},
+	{
+		name: "First",
+		opt: query.IteratorOptions{
+			Expr:      influxql.MustParseExpr("first(value)"),
+			Ascending: true,
+			StartTime: influxql.MinTime,
+			EndTime:   influxql.MaxTime,
+		},
+	},
+	{
+		name: "Last",
+		opt: query.IteratorOptions{
+			Expr:      influxql.MustParseExpr("last(value)"),
+			Ascending: true,
+			StartTime: influxql.MinTime,
+			EndTime:   influxql.MaxTime,
+		},
+	},
+	{
+		name: "Limit",
+		opt: query.IteratorOptions{
+			Expr:      influxql.MustParseExpr("value"),
+			Ascending: true,
+			StartTime: influxql.MinTime,
+			EndTime:   influxql.MaxTime,
+			Limit:     10,
+		},
+	},
+}
+
+var benchmarkVariants = []struct {
+	name   string
+	modify func(opt query.IteratorOptions) query.IteratorOptions
+}{
+	{
+		name: "All",
+		modify: func(opt query.IteratorOptions) query.IteratorOptions {
+			return opt
+		},
+	},
+	{
+		name: "GroupByTime_1m-1h",
+		modify: func(opt query.IteratorOptions) query.IteratorOptions {
+			opt.StartTime = 0
+			opt.EndTime = int64(time.Hour) - 1
+			opt.Interval = query.Interval{
+				Duration: time.Minute,
+			}
+			return opt
+		},
+	},
+	{
+		name: "GroupByTime_1h-1d",
+		modify: func(opt query.IteratorOptions) query.IteratorOptions {
+			opt.StartTime = 0
+			opt.EndTime = int64(24*time.Hour) - 1
+			opt.Interval = query.Interval{
+				Duration: time.Hour,
+			}
+			return opt
+		},
+	},
+	{
+		name: "GroupByTime_1m-1d",
+		modify: func(opt query.IteratorOptions) query.IteratorOptions {
+			opt.StartTime = 0
+			opt.EndTime = int64(24*time.Hour) - 1
+			opt.Interval = query.Interval{
+				Duration: time.Minute,
+			}
+			return opt
+		},
+	},
+	{
+		name: "GroupByHost",
+		modify: func(opt query.IteratorOptions) query.IteratorOptions {
+			opt.Dimensions = []string{"host"}
+			return opt
+		},
+	},
+	{
+		name: "GroupByHostAndTime_1m-1h",
+		modify: func(opt query.IteratorOptions) query.IteratorOptions {
+			opt.Dimensions = []string{"host"}
+			opt.StartTime = 0
+			opt.EndTime = int64(time.Hour) - 1
+			opt.Interval = query.Interval{
+				Duration: time.Minute,
+			}
+			return opt
+		},
+	},
+	{
+		name: "GroupByHostAndTime_1h-1d",
+		modify: func(opt query.IteratorOptions) query.IteratorOptions {
+			opt.Dimensions = []string{"host"}
+			opt.StartTime = 0
+			opt.EndTime = int64(24*time.Hour) - 1
+			opt.Interval = query.Interval{
+				Duration: time.Hour,
+			}
+			return opt
+		},
+	},
+	{
+		name: "GroupByHostAndTime_1m-1d",
+		modify: func(opt query.IteratorOptions) query.IteratorOptions {
+			opt.Dimensions = []string{"host"}
+			opt.StartTime = 0
+			opt.EndTime = int64(24*time.Hour) - 1
+			opt.Interval = query.Interval{
+				Duration: time.Hour,
+			}
+			return opt
+		},
+	},
 }
 
-func benchmarkIterator(b *testing.B, opt query.IteratorOptions, pointN int) {
+func BenchmarkEngine_CreateIterator_Default_1K(b *testing.B) {
+	benchmarkIteratorDefault(b, 1000)
```

Comment:
Why not something like:

```go
func BenchmarkEngine_CreateIterator_Default(b *testing.B) {
    sizes := []int{1000, 100000, 1000000}
    for _, size := range sizes {
		e := MustInitDefaultBenchmarkEngine(size)
		b.Run(fmt.Sprint(size), func(b *testing.B) {
			benchmarkEngine(b, e)
		})
	}
}
```

That way you don't need separate `BenchmarkEngine_CreateIterator_Default_X` functions, and you get nicer output in the results. You will get everything namespaced by `/size/`